input/input.go

```diff
@@ -0,0 +1,264 @@
+/*
+Package Input reads config file and populates the data in Config struct.
+*/
+package input
+
+import (
+	"errors"
+	"os"
+	"strconv"
+
+	"github.com/maplelabs/github-audit/logger"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	DefaultMasterBranch = "master"
+	DefaultMainBranch   = "main"
+	PRIVATE             = "private"
+)
+
+var (
+	ErrMissingAuditJob        = errors.New("no audit job defined")
+	ErrMissingAuditJobName    = errors.New("missing audit job name")
+	ErrMissingTarget          = errors.New("no target defined")
+	ErrMissingAccessToken     = errors.New("missing access token")
+	ErrMissingUsername        = errors.New("missing username")
+	ErrMissingRepositoryName  = errors.New("missing repository name")
+	ErrMissingRepositoryURL   = errors.New("missing repository URL")
+	ErrMissingPollingInterval = errors.New("missing polling interval")
+	ErrMissingRepositoryType  = errors.New("missing repository type")
+	ErrMissingRepositoryHost  = errors.New("missing repository host")
+	ErrMissingRepositoryOwner = errors.New("missing repository owner")
+	ErrMissingTargetNameList  = errors.New("missing target name in audit job")
+	ErrMissingTargetName      = errors.New("missing target name")
+	ErrMissingTargetType      = errors.New("missing target type")
+	ErrPollingIntervalFormat  = errors.New("polling interval format is incorrect")
+)
+
+var (
+	log logger.Logger
+)
+
+func init() {
+	log = logger.GetLogger()
+}
+
+// Config represents the data parsed from config.yaml.
+type Config struct {
+	// Loglevel defines the logging level for the log file. <OPTIONAL>
+	Loglevel string `yaml:"loglevel,omitempty" json:"loglevel,omitempty"`
+	// Logpath represents the path where log file will be generated. <OPTIONAL>
+	Logpath string `yaml:"logpath,omitempty" json:"logpath,omitempty"`
+	// Auditjobs is list of auditjobs. <REQUIRED>
+	AuditJobs []AuditJob `yaml:"auditJobs" json:"auditJobs"`
+	// Targets is list of targets. <REQUIRED>
+	Targets []Target `yaml:"targets" json:"targets"`
+}
+
+// AuditJob represents the auditing job for which data will be fetched from remote git repository.
+type AuditJob struct {
+	// Name of the auditjob. <REQUIRED>
+	Name string `yaml:"name" json:"name"`
+	// Polling interval of the audit job. <REQUIRED> Format: 10s , 10m , 10h , 1d
+	PollingInterval string `yaml:"polling_interval" json:"polling_interval"`
+	// Metadata if any <OPTIONAL>
+	Metadata interface{} `yaml:"metadata" json:"metadata"`
+	// Tags if any <OPTIONAL>
```

Comment:
dont add OPTIONAL or REQUIRED flag here, that is for config file. Same in all structs